airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeInternalStagingConsumerFactory.java

```diff
@@ -45,6 +46,7 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(SnowflakeInternalStagingConsumerFactory.class);
 
   private static final long MAX_BATCH_SIZE_BYTES = 1024 * 1024 * 1024 / 4; // 256mb
+  private static final String CURRENT_SYNC_PATH = UUID.randomUUID().toString();
```

Comment:
I'd prefer for this to be an instance field (i.e. non-static). Then `SnowflakeInternalStagingDestination#getConsumer` would need to call `new SnowflakeInternalStagingConsumerFactory().create(...)`

generally I avoid non-constant static fields, since they can be confusing + difficult to test.

Reply:
_CURRENT_SYNC_PATH_ field used in static methods _recordWriterFunction_ and _onCloseFunction_. Non-static field can't be used in static methods

Reply:
I think those methods could just be switched to non-static as well (they're only called from `create`, so that should be safe)

Reply:
@edgao switched to non-static